### ISSUE TITLE
Create a Linux version

### ISSUE THREAD

**Issue by sbstn87** [digimezzo/dopamine-windows#810]
## I'm submitting a...

<pre><code>
[ ] Regression 
[ ] Bug report  
[x ] Feature request
</code></pre>

Is it possible to make a Linux version of Dopamine?

**Comment by PhillyMay** [comment]
Woud like to see a Linux Version too. Preferably a distribution agnostic version (AppImage/Flatpack/Snap), so that everyone is able to use it. 

I really like Solus OS because it does a lot of important things right (rolling release, beautiful ui, clean and up2date "app center", etc.). The problem with such distributions is, that they are not based on anything else (like Ubuntu = Debian or Manjaro = Arch), so the app selection is not as good as on some other distros. Plattform agnostic packages would help minimizing the fragmentation of Linux, without sacrificing the idea of a having different distros for different use cases (Tablet, Server, Desktop, etc)

**Comment by sbstn87** [comment]
+1 for a Flatpak release

**Comment by digimezzo** [comment]
What are the current GUI frameworks for Linux? Is it still GTK and QT? (I haven't used Linux in years).
If there is a UI modern, usable and stable UI framework that supports .NET/Mono, I might work a Linux version. WPF is what makes Dopamine amazing. I'm, however, afraid there isn't such thing on Linux.

**Comment by sbstn87** [comment]
Yeah, GTK and Qt still are the dominant GUI frameworks. In my opinion Qt is currently the one that offers better look and feel.

I noticed that WPF is XML based. For Qt there's something called QML that may offer similar possibilities:
https://en.wikipedia.org/wiki/QML

The official KDE music player is based on QML AFAIK:
https://vvave.kde.org/

**Comment by Magentaize** [comment]
Impossible, WPF is a 2D/3D graphic engine based on DirectX and *nix doesn't has a perfect engine except Valve Source.

**Comment by digimezzo** [comment]
I, more and more, want to do this. 
I've been a Linux users for years, before I moved back to Windows a few years ago. Today, I've been looking which are the current audio players on Linux. Sadly, things haven't changed at all since so long ago. So, I'll try to find some time to look at this. The Linux version won't be exactly like the Windows version (due to framework differences), but I'll try to keep the philosophy from the Windows version.

Avalonia UI looks promising; http://avaloniaui.net

**Comment by Riajyuu** [comment]
@digimezzo I afraid you have to rewrite most things. plus, too many desktop environment to be adapted on Linux.

**Comment by sbstn87** [comment]
@digimezzo That would be great. Dopamine could be the best looking music player on Linux. Please keep us updated on that.

**Comment by digimezzo** [comment]
@Rictusempra I realize that. It will be a big challenge. But I'm willing to give it a try.

**Comment by digimezzo** [comment]
I've looked at the possiblities for UI development in Linux. At some point I considered Electron, to be able to provide a UI experience like on Windows. With web tech it would be possible to use a style similar to the Windows Dopamine. I, however, think Linux users would prefer a more native experience. 
So my current preference lies in creating a Python/GTK+3 version of Dopamine. The idea is that it would offer a UI layout similar to the Windows Dopamine, but using a more GTK native experience. It would stick to the selected GTK theme.
What do you think about this?

**Comment by Magentaize** [comment]
Rewriting Dopamine into another language will be a magnificent picture :)

**Comment by PrinceOwen9466** [comment]
Hey check this out, (http://avaloniaui.net) you can basically create a linux and mac version of Dopamine using XAML and Mono! I have really enjoyed how awesome and incredible Dopamine has been, It would definitely be a dream come true to have such a great music player on linux. I'm a .Net developer myself, I would love to contribute if you need some help!

**Comment by trymeouteh** [comment]
I would like to suggest Qt framework. I heard it is a great UI for making desktop applications.

If a Linux app is released then I would ditch my other linux music players for this.

**Comment by AldyTS** [comment]
no linux version yet?

**Comment by digimezzo** [comment]
@AldyTS I'm working on it. My time has however been very limited lately :( But it is in the works.

**Comment by AldyTS** [comment]
@digimezzo thanks for responding! I'm glad you didn't abandon the idea dopamine music player on linux, because dopamine is my go to music player when I'm using windows and I've already tried multiple music players on linux and they just can't be compared to dopamine in terms of ease of use and sleekness!

**Comment by NNois** [comment]
Oh this would be so cool.
On linux music players are so old style, we need Spotify like ones, so straightforward
